libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -897,6 +899,71 @@ std::string YulUtilFunctions::resizeDynamicArrayFunction(ArrayType const& _type)
 	});
 }
 
+string YulUtilFunctions::resizeDynamicByteArrayFunction(ArrayType const& _type)
+{
+	string functionName = "resize_array_" + _type.identifier();
+	return m_functionCollector.createFunction(functionName, [&]() {
+		return Whiskers(R"(
+			function <functionName>(array, newLen) {
+				if gt(newLen, <maxArrayLength>) {
+					<panic>()
+				}
+
+				let oldLen := <fetchLength>(array)
+
+				if gt(newLen, oldLen) {
+					sstore(array, or(shl(newLen, 1), iszero(lt(newLen, 32))))
+				}
+
+				// Size was reduced, clear end of array
+				if lt(newLen, oldLen) {
+					let offset := mod(newLen, 32)
+					let oldSlotCount := add(iszero(oldLen), div(add(oldLen, 31), 32))
```

Comment:
Why is the slot count 1 for length zero?

Reply:
I will fix it